File: IntentExample/app/src/main/java/com/tlu/intentexample/MainActivity.java
package com.tlu.intentexample;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import java.util.Objects;

public class MainActivity extends AppCompatActivity {
    private Button btn_request;
    private EditText txt_a, txt_b, txt_kq;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
        btn_request = findViewById(R.id.btn_main);
        txt_a = findViewById(R.id.txt_a);
        txt_b = findViewById(R.id.txt_b);
        txt_kq = findViewById(R.id.txt_kq);

        btn_request.setOnClickListener(v -> {
            Toast.makeText(getApplicationContext(), "adasdasd", Toast.LENGTH_SHORT).show();

            Intent intent = new Intent(MainActivity.this, MainActivity2.class);
            Bundle bundle = new Bundle();
            bundle.putInt("txt_a", Integer.parseInt("0" + txt_a.getText()));
            bundle.putInt("txt_b", Integer.parseInt("0" + txt_b.getText()));
            intent.putExtra("main_data", bundle);
            startActivityForResult(intent,99);
        });


    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (resultCode == 33 && requestCode == 99) {
            txt_kq.setText(String.valueOf(data.getIntExtra("kq", 0)));
        }
        if(resultCode == 34 && requestCode == 99){
            txt_kq.setText(String.valueOf(data.getIntExtra("kq", 0)));
        }

    }
}

File: Intentcallsms/app/src/main/java/com/tlu/intentcallsms/MainActivity.kt
package com.tlu.intentcallsms

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {


    private lateinit var btn_call: Button
    private lateinit var btn_sms: Button
    private lateinit var btnCamera: Button
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
        init()


        btn_call.setOnClickListener(
            ::onClickButton
        )

        btn_sms.setOnClickListener(
            ::onClickButton
        )
        btnCamera.setOnClickListener(
            ::onClickButton
        )
    }

    private fun onClickButton(view: View) {
        when (view.id) {
            R.id.btn_call -> {
                val intentPhone = Intent(this,CallActivity::class.java)
                startActivity(intentPhone)
            }

            R.id.btn_sms -> {
                val intentSMS= Intent(this,SmsActivity::class.java)
                startActivity(intentSMS)
            }
            R.id.btn_camera -> {
                val intentCamera= Intent(this,CameraActivity::class.java)
                startActivity(intentCamera)
            }
        }
    }

    private fun init() {
        btn_call = findViewById(R.id.btn_call)
        btn_sms = findViewById(R.id.btn_sms)
        btnCamera = findViewById(R.id.btn_camera)
    }
}